executor/distsql.go

```diff
@@ -292,19 +292,24 @@ type TableReaderExecutor struct {
 	// columns are only required by union scan.
 	columns []*model.ColumnInfo
 
-	// result returns one or more distsql.PartialResult and each PartialResult is returned by one region.
-	result        distsql.SelectResult
+	// partialResult store the result from one region.
 	partialResult distsql.PartialResult
+	// resultHandler handles the order of the result. Since (MAXInt64, MAXUint64] stores before [0, MaxInt64] physically
+	// for unsigned int.
+	resultHandler *tableResultHandler
 	priority      int
 	feedback      *statistics.QueryFeedback
 }
 
 // Close implements the Executor Close interface.
 func (e *TableReaderExecutor) Close() error {
-	e.feedback.SetIntRanges(e.ranges).SetActual(e.result.ScanCount())
+	e.feedback.SetIntRanges(e.ranges).SetActual(e.resultHandler.totalCount())
 	e.ctx.StoreQueryFeedback(e.feedback)
-	err := closeAll(e.result, e.partialResult)
-	e.result = nil
+	err := e.resultHandler.close()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	err = closeAll(e.partialResult)
```

Comment:
If the `resultHandle` return error, then `partialResult` will not be closed? Why not just capitalize the `close` for `resultHandler` and use `closeAll`?